NOTICE.md

```diff
@@ -26,36 +26,36 @@ The project maintains the following source code repositoriy:
 
 * https://github.com/eclipse-edc/Connector
 
-## Third-party Content (Overarching All Modules)
+## Third Party Content
```

Comment:
The hyphen should be kept as third-party acts as adjective to content. I agree to use a capital letter for "Party" though.